src/rez/package_order.py

```diff
@@ -608,6 +609,265 @@ def from_pod(cls, data):
+    Specified through the "packages" attributes, which should be a dict which
+    maps from a package family name to a list of version ranges to prioritize,
+    in decreasing priority order.
+    As an example, consider a package splunge which has versions:
+      [1.0, 1.1, 1.2, 1.4, 2.0, 2.1, 3.0, 3.2]
+    By default, version priority is given to the higest version, so version
+    priority, from most to least preferred, is:
+      [3.2, 3.0, 2.1, 2.0, 1.4, 1.2, 1.1, 1.0]
+    However, if you set a custom package order like this:
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: ['2', '1.1+<1.4']
+    Then the preferred versions, from most to least preferred, will be:
+     [2.1, 2.0, 1.2, 1.1, 3.2, 3.0, 1.4, 1.0]
+    Any version which does not match any of these expressions are sorted in
+    decreasing version order (like normal) and then appended to this list (so they
+    have lower priority). This provides an easy means to effectively set a
+    "default version."  So if you do:
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: ['3.0']
+    resulting order is:
+      [3.0, 3.2, 2.1, 2.0, 1.4, 1.2, 1.1, 1.0]
+    You may also include a single False or empty string in the list, in which case
+    all "other" versions will be placed at that spot. ie
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: ['', '3+']
+    yields:
+     [2.1, 2.0, 1.4, 1.2, 1.1, 1.0, 3.2, 3.0]
+    Note that you could also have gotten the same result by doing:
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: ['<3']
+    If a version matches more than one range expression, it will be placed at
+    the highest-priority matching spot, so:
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: ['1.2+<=2.0', '1.1+<3']
+    gives:
+     [2.0, 1.4, 1.2, 2.1, 1.1, 3.2, 3.0, 1.0]
+    Also note that this does not change the version sort order for any purpose but
+    determining solving priorities - for instance, even if version priorities is:
+      package_orderers:
+      - type: custom
+        packages:
+          splunge: [2, 3, 1]
+    The expression splunge-1+<3 would still match version 2.
+    """
+    name = "custom"
+
+    def __init__(self,
+                 packages: Dict[str, List[Union[str, VersionRange]]],
+                 version_orderer: Optional[Union[PackageOrder, Dict[str, Any]]] = None,
+                 ):
+        """
+        Args:
+            packages: (Dict[str, List[VersionRange]]): packages that
+                this orderer should apply to, and the version priority ordering
+                for that package
+            version_orderer (Optional[Union[PackageOrder, Dict[str, Any]]]):
+                How versions are sorted within version ranges.
+                Can take a pod representation of an orderer or a PackageOrder object
```

Comment:
this seems like a nice solution to the chaining problem.    Any downsides that we should document?

Reply:
No, it's just specific to the custom package order.  Would be nice to be able to arbitrarily chain them together in a general way.